src/Symfony/Component/Form/Extension/Core/Type/TimeType.php

```diff
@@ -57,16 +57,18 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         if ('single_text' === $options['widget']) {
             $builder->addViewTransformer(new DateTimeToStringTransformer($options['model_timezone'], $options['view_timezone'], $format));
 
-            // handle seconds ignored by user's browser when with_seconds enabled
-            // https://codereview.chromium.org/450533009/
-            if ($options['with_seconds']) {
-                $builder->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $e) {
-                    $data = $e->getData();
-                    if ($data && preg_match('/^\d{2}:\d{2}$/', $data)) {
-                        $e->setData($data.':00');
+            $builder->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $e) use ($options) {
+                $data = $e->getData();
+                if ($data && preg_match('/^(?P<hours>\d{2}):(?P<minutes>\d{2})(?::(?P<seconds>\d{2})\.0{3})?$/', $data, $matches)) {
```

Comment:
do we need to support missing milliseconds?
https://tools.ietf.org/html/rfc3339#page-8 looks like the end could be `(?:\.0{1,2})?`?

Reply:
I didn't account for that as I understood that Edge would always send milliseconds. But being even less restrictive might be a good idea to account for further changes in browsers.

Reply:
code updated and test added